app/models/customer.rb

```diff
@@ -158,6 +159,11 @@ def self.current_customer(reset = false)
     @current_customer[Apartment::Tenant.current] ||= Customer.where(:url_shortcut => Apartment::Tenant.current).first || Customer.new
   end
   
+  # Unset the customer attribute in the current thread. Used as an +after_save+ callback to pick up changes.
+  def reset_customer_in_thread
+    Thread.current['customer'] = nil
```

Comment:
This is strictly better than what we have right now, but I'm not sure it's a complete fix.

Let's say we use a multithreaded server, or there are some Sidekiq threads sitting around that already have their `customer` set. This seems like fragile behavior in that only the single thread where the current customer is set will be refreshed, while others will have a stale cache.

I don't know enough about how nginx/unicorn recycles worker threads to know how big an issue this will be in production. We almost never edit customers, so I don't think it's merge blocking, but it's something we should be watching going forward.